applications: Asset Tracker_v2: Split `prj.conf` into multiple overlays

simensrostad opened the pull request:
Split `prj.conf` into multiple overlay configuration files where
each file contains options that are specific to a feature.

To build with a specific feature the corresponding overlay file
needs to be included in the `OVERLAY_CONFIG` CMake list-variable when
building the application.

This patch is to provide a more modular approach compared to having all
configurations in the root `prj.conf` and relying on the user to
comment out/in options when wanting to enable a feature.

CIA-390 CIA-407

coderbyheart commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -410,7 +414,12 @@ The following configuration files are available in the application folder:
 * :file:`boards/nrf9160dk_nrf9160_ns.conf` - Configuration file specific for nRF9160 DK. This file is automatically merged with the :file:`prj.conf` file when you build for the ``nrf9160dk_nrf9160_ns`` build target.
-* :file:`overlay-low-power.conf` - Configuration file that achieves the lowest power consumption by disabling features  that consume extra power like LED control and logging.
+* :file:`overlay-nrf_cloud.conf` - Configuration file to set nRF Cloud as cloud provider.
+* :file:`overlay-aws.conf` - Configuration file to set AWS IoT as cloud provider.
```
```suggestion
* :file:`overlay-aws.conf` - Configuration file to set AWS as cloud provider.
```

coderbyheart commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -410,7 +414,12 @@ The following configuration files are available in the application folder:
-* :file:`overlay-low-power.conf` - Configuration file that achieves the lowest power consumption by disabling features  that consume extra power like LED control and logging.
+* :file:`overlay-nrf_cloud.conf` - Configuration file to set nRF Cloud as cloud provider.
+* :file:`overlay-aws.conf` - Configuration file to set AWS IoT as cloud provider.
+* :file:`overlay-azure.conf` - Configuration file to set Azure IoT Hub as cloud provider.
```
```suggestion
* :file:`overlay-azure.conf` - Configuration file to set Azure as cloud provider.
```

coderbyheart commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -428,6 +437,11 @@ Building and running
 Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.
 
+The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command. `OVERLAY_CONFIG` contains a set
+of entries that enables certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,
```
```suggestion
of entries that enable certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,
```

peknis replied:
-> a set of entries that enables

coderbyheart commented on applications/asset_tracker_v2/prj.conf:
```diff
@@ -166,7 +86,7 @@ CONFIG_EVENT_MANAGER=y
 CONFIG_EVENT_MANAGER_LOG_EVENT_TYPE=n
 
-# cJSON - Used in AWS FOTA and cloud data traffic encoding.
+# cJSON - Used in cloud data traffic encoding.
```
```suggestion
# cJSON - Used in cloud data encoding.
```

coderbyheart commented on applications/asset_tracker_v2/src/cloud/azure_iot_hub_integration.c:
```diff
@@ -9,6 +9,13 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_CLOUD_INTEGRATION_LOG_LEVEL);
 
+BUILD_ASSERT(IS_ENABLED(CONFIG_AGPS_SRC_SUPL) == 0,
```
Could be a KConfig dependency instead.

simensrostad replied:
With the current implementation this would create a dependency in lib `azure_iot_hub` to lib `supl` . We currently switch in cloud integrations based on the Kconfig that enable the specific cloud libs. `CONFIG_AZURE_IOT_HUB` for instance, which reside in `subsys/net/lib/azure_iot_hub/kconfig`

We could look into creating app level kconfigs for switching in cloud integration layers, then we could add these Kconfig dependencies without messing with sub libraries.

coderbyheart commented on applications/asset_tracker_v2/src/cloud/azure_iot_hub_integration.c:
```diff
@@ -9,6 +9,13 @@
+	     "The Azure IoT Hub integration only supports A-GPS over SUPL. Enable SUPL by setting "
+	     "CONFIG_AGPS_SRC_SUPL");
+
+BUILD_ASSERT(IS_ENABLED(CONFIG_NRF_CLOUD_PGPS) == 0,
```
This should be expressible as well in KConfig, CONFIG_AZURE_IOT_HUB=y depends on CONFIG_NRF_CLOUD_PGPS=n

simensrostad replied:
Same goes for this one.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-5662
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -440,27 +454,29 @@ Also, the device must be provisioned and configured with the certificates accord
+   west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-low_power.conf
 
-The above command will build for nRF9160 DK using the configurations found in :file:`overlay-low-power.conf`, in addition to the configurations found in :file:`prj_nrf9160dk_nrf9160_ns.conf`.
+The above command will build for nRF9160 DK using the configurations found in :file:`overlay-low_power.conf`, in addition to the configurations found in :file:`prj_nrf9160dk_nrf9160_ns.conf`.
```
```suggestion
The above command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low_power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -428,6 +437,11 @@ Building and running
 Before building and running the firmware ensure that the cloud side is set up.
 Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.
 
+The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command. `OVERLAY_CONFIG` contains a set
+of entries that enables certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,
```
```suggestion
The application requires that a CMake list-variable `OVERLAY_CONFIG` is set before calling the west build command.
The `OVERLAY_CONFIG` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
If the `OVERLAY_CONFIG` CMake variable is not set,
```

peknis replied:
Remember, each sentence starts a new line.

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -428,6 +437,11 @@ Building and running
 
+The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command. `OVERLAY_CONFIG` contains a set
+of entries that enables certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,
+the application will default to using nRF Cloud as the cloud provider. See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the
```
```suggestion
the application defaults to using nRF Cloud as the cloud provider.
See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the
```

tejlmand commented on applications/asset_tracker_v2/CMakeLists.txt:
```diff
@@ -6,6 +6,11 @@
 
+if(NOT DEFINED OVERLAY_CONFIG)
+  message(WARNING "Overlay configuration list is empty, using nRF Cloud as the default cloud provider")
+  list(APPEND OVERLAY_CONFIG "overlay-nrf_cloud.conf")
```
we should not apply any config fragments per default.

Users can create a custom `my_conf.conf` and use it with a sample like: `cmake ... -DCONF_FILE=/<path>/my_conf.conf` and then build the sample with known set of settings.

But in this app you would apply an additional config fragment begind user's back in such case.

That is not acceptable.

jhn-nordic replied:
Do you have a proposal on how to solve this? We want the user to have a functional sample when building without options.

tejlmand replied:
> Board-specific config files are also patched in behind the user's "back".
> How is that different?

Actually board specific config files are not patched behind users back.

The board fragments is a documented feature, and it only applies when default `prf.conf` files are used:
https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#the-initial-configuration
Or a `prj_<build>.conf` is in use, in which case the `boards/<board>_<build>.conf` is used, if exists.

Test for yourself:
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/subsys/settings
...
Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
Merged configuration '/projects/github/ncs/zephyr/samples/subsys/settings/prj.conf'
Merged configuration '/projects/github/ncs/zephyr/samples/subsys/settings/boards/nrf52840dk_nrf52840.conf'
Configuration saved to '/projects/github/ncs/zephyr/build/zephyr/.config'
```

but when using a custom conf file, the board file is NOT used:
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/subsys/settings -DCONF_FILE=myconftest.conf
...
Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
Merged configuration 'myconftest.conf'
Configuration saved to '/projects/github/ncs/zephyr/build/zephyr/.config'
```
Notice, that with `-DCONF_FILE=myconftest.conf` then `boards/nrf52840dk_nrf52840.conf'` is no longer used.

> Isn't that sufficient enough?

you are still deviating on how things are generally working.
We can't expect users to read each and every user guide in details to understand exactly how this sample decides to add extra configurations. And then another sample does things a bit differently.

Secondly, user's cannot avoid `overlay-nrf_cloud.conf` to be applied with this PR, unless a dummy / empty fragment is created and applied.

tejlmand replied:
> o you have a proposal on how to solve this? We want the user to have a functional application when building without options. And at the same time we do not want them to comment out several config options if they want to build with another cloud vendor backend. (nrfcloud(default), aws and azure))

Let me take a closer look at the sample.
One question, is it allowed to select multiple cloud vendor backends, like both nrfcloud and aws, or are they mutual exclusive ?

coderbyheart replied:
Mutual exclusive.

simensrostad replied:
Alright, thanks for the explanation @tejlmand. Yes, they are mutual exclusive. The application requires that either `overlay-nrf_cloud.conf`, `overlay-azure.conf`, or `overlay-aws.conf` be configured with the application. The rest is the overlays are optional.

SebastianBoe replied:
IMHO the originally proposed PR is acceptable.

But if it must be changed such that OVERLAY_CONFIG is not set by the app build scripts then we could do

if(NOT DEFINED CONF_FILE)
set(CONF_FILE prj.conf overlay-nrf_cloud.conf)
endif()

and instruct the user to use a list of CONF_FILE instead of OVERLAY_CONFIG.

Is this acceptable @tejlmand, or do you have a better suggestion perhaps ?

tejlmand replied:
> However, the nRF Cloud library suite has other means of communications with nRF Cloud such as via REST. Then CONFIG_NRF_CLOUD_MQTT isn't needed.

Yes, but a default doesn't mean it can't be disabled.
Of course users not wanting to use `MQTT` would then have to opt-out, like:
```
CONFIG_NRF_CLOUD=y
CONFIG_NRF_CLOUD_MQTT=n
```
but if `MQTT` is a good default it will allow majority of samples to simply do:
```
CONFIG_NRF_CLOUD=y
```
and forget about `MQTT`.
Also reducing number of lines in our cloud sample config files.

I'll leave the decision on this for you guys, just mentioning my observations.

> If this option is enabled, the socket is exposed and its expected that the polling is done in the application.

Did you perhaps mean `If this option is disabled` ?

> >  Same regarding the setting: CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD=y.
>
> This option can probably default to y. AFAIK there are no apps/samples that uses nRF Cloud MQTT that polls in the application.

It sounds to me this should have a `default y`.
Remember, samples can always disable this with `CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD=n`, and sane default values make life easier for majority of users, and reduces number of lines in our config files.

> so why not have it defaulting to GPS_MODULE_NMEA when NRF_CLOUD is selected ?
>
> Originally, the idea was to keep the GPS module as generic as possible so that it could be easily be reused in other applications and not have artifacts from Kconfigs options / functionality from non-related libraries.

In that case you should give the choice a variable name, as that will allow to set the default in another Kconfig file, like:
```
choice GPS_MODULE_DATA_FORMAT
	default GPS_MODULE_NMEA if NRF_CLOUD
endchoice
```
and then in the current Kconfig code for the GPS, you would have:
```
choice GPS_MODULE_DATA_FORMAT
	prompt "Select GNSS data format"
	default GPS_MODULE_PVT
...
endchoice
```
That way you may keep knowledge where it rightfully belongs.

Somewhere I assume there must be a dependency.
Either GPS knows about cloud service (probably not).
Or a cloud service knows something about GPS (it seems nRF Cloud has some sort of expectation on GPS for positioning feature)
Or a dedicated Kconfig that glues such knowledge together.

As long as the GPS module is application specific, the `<app>/Kconfig` file sounds like the right location for this.
When / if GPS module is taken out as a generic module, then it can be considered where such code should be placed.

tejlmand reviewed (changes requested):
Config fragments with `OVERLAY_CONFIG` should not be applied automatically.

simensrostad commented:
@coderbyheart @peknis Review comments addressed, thanks!

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -410,7 +414,12 @@ The following configuration files are available in the application folder:
+* :file:`overlay-azure.conf` - Configuration file to set Azure as cloud provider.
+* :file:`overlay-agps.conf` - Configuration file to enable A-GPS.
+* :file:`overlay-pgps.conf` - Configuration file to enable P-GPS.
+* :file:`overlay-low_power.conf` - Configuration file that achieves the lowest power consumption by disabling features that consume extra power like LED control and logging.
```
```suggestion
* :file:`overlay-low_power.conf` - Configuration file that achieves the lowest power consumption by disabling features that consume extra power, such as LED control and logging.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -428,6 +437,11 @@ Building and running
 Before building and running the firmware ensure that the cloud side is set up.
 Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.
 
+The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command.
+The `OVERLAY_CONFIG` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
+If the `OVERLAY_CONFIG` CMake variable is not set, the application defaults to using nRF Cloud as the cloud provider.
+See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the same time.
```
```suggestion
The application requires that you have set the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
The ``OVERLAY_CONFIG`` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
If the you have not set the ``OVERLAY_CONFIG`` CMake variable, the application defaults to using nRF Cloud as the cloud provider.
See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the same time.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -440,27 +454,29 @@ Also, the device must be provisioned and configured with the certificates accord
 
 .. note::
-   For nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_NRF9160_GPS_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
+   For nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
```
```suggestion
   For the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
```

simensrostad commented:
@peknis Suggestions added.

simensrostad commented:
This PR is more or less in a finished state at this point. Taking it out of draft.

simensrostad commented:
This PR now contains cleanups related to lib nRF cloud Kconfig options. Most notably, removal of the `CONFIG_NRF_CLOUD` option. This was necessary to be able to impose a choice selection on transport libraries that is considered mutual exclusive by the application. So that the various MQTT transport libraries can be switched when building with corresponding overlays. The functionality that enables this is added in https://github.com/nrfconnect/sdk-nrf/pull/5662/commits/19470456cf6d6da2899d69f332e60d6e98f934a1.

jayteemo commented on subsys/net/lib/Kconfig:
```diff
@@ -5,6 +5,27 @@
+	bool
+	default y if "$(CLOUD_SERVICE_SELECTOR)" = "choice"
+	help
+	  Internal setting which indicates that the Cloud Service are mutual
```
```suggestion
	  Internal setting which indicates that the Cloud Service is mutually
```

jayteemo commented on subsys/net/lib/Kconfig:
```diff
@@ -5,6 +5,27 @@
+	  Internal setting which indicates that the Cloud Service are mutual
+	  exclusive.
+	  An application defconfig can indicate that the Cloud Service are
+	  mutual exclusive by requesting that selection of cloud services should
```
```suggestion
	  mutually exclusive by requesting that selection of cloud services should
```

jayteemo commented on subsys/net/lib/Kconfig:
```diff
@@ -5,6 +5,27 @@
+	help
+	  Internal setting which indicates that the Cloud Service are mutual
+	  exclusive.
+	  An application defconfig can indicate that the Cloud Service are
```
```suggestion
	  An application defconfig can indicate that the Cloud Service is
```

jayteemo commented on subsys/net/lib/nrf_cloud/Kconfig:
```diff
@@ -3,11 +3,7 @@
-	select CJSON_LIB
-
-if NRF_CLOUD
+if NRF_CLOUD_MQTT || NRF_CLOUD_REST || NRF_CLOUD_PGPS
```
what is the case where `NRF_CLOUD_PGPS` needs the host name or sec tag, but MQTT or REST is not already enabled?

simensrostad replied:
When the application is configured for either AWS IoT or Azure IoT Hub. In those cases we are not dependent on the nRF Cloud MQTT transport to receive/request A/P-GPS data. I included NRF_CLOUD_PGPS to avoid build error when including the nRF Cloud PGPS library.

coderbyheart commented on applications/asset_tracker_v2/overlay-agps.conf:
```diff
@@ -0,0 +1,18 @@
+CONFIG_AGPS_SRC_NRF_CLOUD=y
+CONFIG_NRF_CLOUD_AGPS=y
+
+# A-GPS using SUPL server.
+# Due to licensing, the SUPL library must be manually downloaded and configured
+# according to the instructions found in the SUPL client library documentation in NCS.
+# CONFIG_AGPS_SRC_SUPL=y
```
`CONFIG_AGPS_SRC_SUPL` was removed in #5882

tokangas replied:
The A-GPS configuration is for the A-GPS library, which is not used anymore. Please check the current configuration and update accordingly. Enabling `CONFIG_NRF_CLOUD_AGPS` should be enough. The comment has been updated a bit as well.

@simensrostad But, is it intentional that A-GPS is not used by default like before? Now the application has to be built with the A-GPS overlay to enable A-GPS support. Is that what we want?

simensrostad replied:
Thanks! @tokangas Good point. We should have A-GPS enabled by default. I don't see any benefits of keeping `overlay-agps.conf` in that case. Ill remove it if there aren't any objections.

tokangas commented on applications/asset_tracker_v2/prj.conf:
```diff
@@ -49,83 +49,12 @@ CONFIG_LTE_PSM_REQ_RAT="00001010"
-CONFIG_NRF_CLOUD_SEND_TIMEOUT=y
-
-# GPS
-CONFIG_GPS_MODULE_NMEA=y
```
`CONFIG_GPS_MODULE_NMEA` doesn't seem to be enabled anywhere anymore. It is needed with nRF Cloud, I think.

simensrostad replied:
`GPS_MODULE_NMEA` is the default if building with `NRF_CLOUD_MQTT`.
See the changes in applications/asset_tracker_v2/src/modules/Kconfig.gps_module

simensrostad commented:
@jayteemo @tokangas @coderbyheart Review comments addressed.
- Removed `overlay-agps.conf`. A-GPS is now a default feature for all builds.
- Updated `Kconfig.cloud_module` to set more default values that are common across all builds.
- Updated `asset_tracker_v2_description.rst` to reflect the aforementioned changes. @peknis I think you need have a second look. There has been minor changes to the docs since you last approved.

tejlmand reviewed (approved):
LGTM 👍 
Thanks a lot for the effort.

trantanen commented on subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_mqtt:
```diff
@@ -3,14 +3,22 @@
 
-menuconfig NRF_CLOUD_MQTT
-	bool "nRF Cloud MQTT"
+module = NRF_CLOUD_MQTT
```
Just one question as I haven't been able to look all the details of this PR.
Can `CONFIG_NRF_CLOUD_MQTT` and `CONFIG_NRF_CLOUD_REST` still be turned on simultaneously?

simensrostad replied:
Yes. That is still possible.

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -425,6 +429,12 @@ Building and running
 Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.
 
+The application defaults to using nRF Cloud as the cloud provider.
+However, this can be changed by specifying the overlay file that corresponds to another supported cloud provider when building the application.
+This is done by setting the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
```
```suggestion
However, you can change the cloud provider by specifying the overlay file that corresponds to another supported cloud provider when building the application.
Set the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -425,6 +429,12 @@ Building and running
+The application defaults to using nRF Cloud as the cloud provider.
+However, this can be changed by specifying the overlay file that corresponds to another supported cloud provider when building the application.
+This is done by setting the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
+The ``OVERLAY_CONFIG`` CMake variable can be set to contain a list of overlays that will be patched in, each enabling a specific feature.
```
```suggestion
You can set this variable to contain a list of overlays that will be patched in, each enabling a specific feature.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -437,10 +447,12 @@ Also, the device must be provisioned and configured with the certificates accord
 
 .. note::
-   For nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_NRF9160_GPS_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
+   For the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
```
```suggestion
   When you build the application for the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` to achieve the best external antenna performance.
```

peknis commented on applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:
```diff
@@ -450,14 +462,14 @@ To build with Kconfig overlay, it must be based to the build system, as shown in
    west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-low-power.conf
 
-The above command will build for nRF9160 DK using the configurations found in :file:`overlay-low-power.conf`, in addition to the configurations found in :file:`prj_nrf9160dk_nrf9160_ns.conf`.
+The above command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low-power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.
```
```suggestion
This command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low-power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.
```

peknis commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -59,6 +59,9 @@ nRF9160: Asset Tracker v2
 * Updated the application to start sending neighbor cell measurement data to nRF Cloud.
 * Added content-type and encoding properties to outgoing Azure MQTT messages.
 * Added support for A-GPS and P-GPS in Azure IoT Hub integration.
+* Added new overlay configuration files and split the :file:`prj.conf` file into the various files that now bind specific features.
```
```suggestion
* Added new overlay configuration files and split the :file:`prj.conf` file to several files that now bind specific features.
```

peknis commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -325,6 +328,7 @@ Libraries for networking
 
 * :ref:`lib_nrf_cloud` library:
 
+  * Removed the `CONFIG_NRF_CLOUD` Kconfig option.
```
```suggestion
  * Removed the ``CONFIG_NRF_CLOUD`` Kconfig option.
```
